api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -126,10 +123,13 @@ public User createUser(String givenName, String familyName, String username, Str
       .setEmails(new UserEmail().setType("custom").setAddress(contactEmail).setCustomType("contact"))
       .setChangePasswordAtNextLogin(true);
     retryHandler.run((context) -> getGoogleDirectoryService().users().insert(user).execute());
-    //This is a temporary solution until Mandrill is implemented.
-    //This way if one merge happens and the other is delayed for any reason,
-    //we won't break in a major way.
-    sendPasswordEmail(contactEmail, password, user);
+    try {
+      mailServiceProvider.get().sendWelcomeEmail(contactEmail, password, user);
+    } catch (MessagingException e) {
+      log.log(Level.SEVERE, String.format(
+        "Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, username, e.toString()));
+      throw new WorkbenchException(e);
```

Comment:
So let's say Mandrill is down temporarily, and this request fails. The user is now in a bit of a pickle; they can't log in with their new account without having received an email with their credentials.

One way of mitigating this is doing retries here or inside sendWelcomeEmail, with some backoff. Another option (more work, probably not worth it now?) is to have the email be sent in a background task asynchronously that can be retried, so that longer outages of Mandrill can still be recovered from.

Anyway, we may want to do something here.

Reply:
Going to add in some retries. Also going to move logging into MailService. Still up for conversation on how to handle longer outages...